internal/controller/providers/pxc/applier.go

```diff
@@ -117,6 +173,20 @@ func (p *applier) Engine() error {
 	}
 
 	pxc.Spec.SecretsName = p.DB.Spec.Engine.UserSecretsName
+	// XXX: PXCO v1.17.0 has a bug in the auto generation of the secrets where
+	// there's a 1.5% chance that the proxyadmin password starts with the '*'
+	// character, which will be rejected by the secret validation, leaving the
+	// cluster stuck in the 'initializing' state.
+	// See: https://perconadev.atlassian.net/browse/K8SPXC-1568
+	// To work around this, we need to set the password ourselves to bypass the
+	// buggy auto generation.
+	if p.DBEngine.Status.OperatorVersion == "1.17.0" {
+		err := p.ensureSecretHasProxyAdminPassword(pxc.Spec.SecretsName)
+		if err != nil {
+			return fmt.Errorf("failed to set proxyadmin password: %w", err)
+		}
+	}
```

Comment:
Just wanted to highlight that with this workaround:
- we assume that in 1.18.0 the problem will be fixed. So far the fix is indeed planned for 1.18.0. 
- we accept that the user can't change this password. Not that Everest was explicitly allowing that before but technically there was still an option to change the password in secrets. With this workaround the change won't take any effect.

This is a non-blocking comment, feel free to resolve it.

Reply:
I confirmed with Slava that this will be fixed in 1.18.0.
Why wouldn't the user be able to change this password? We are only generating a password if it doesn't exist in the secret already. We will not forbid a password change nor will we replace an existing one. Am I missing something?

Reply:
I got under impression that with `CreateOrUpdateSecretData` we're replacing the existing password which we actually don't. I missed the part where we check wether the password is already there and do nothing in that case. So never mind the second note, sorry for the confusion.